### ISSUE TITLE
Commit count metric

### ISSUE THREAD

**Issue by rjaiwal5139** [mvisonneau/gitlab-ci-pipelines-exporter#51]
Hello, 
           Is it possible to add a metric for commit count per repo since the last build? 

Thanks

**Comment by mvisonneau** [comment]
not sure how to get this information from the API 🤔

**Comment by rjaiwal5139** [comment]
Maybe from here: https://docs.gitlab.com/ee/api/commits.html

**Comment by DJMatus23** [comment]
There is a `commit_count` as part of the projects API https://docs.gitlab.com/ee/api/projects.html - I was thinking of adding a PR for gathering these stats.  You probably don't want to add them under `gitlab_ci_pipeline_` as they are stats of the git repo, so you'd be better to do something like `git_repo_commit_count` or something like that.

I was hoping I could find another exporter that already had this in there.  Ideally something generic that provided some git stats taken from a number of git provider APIs, including gitlab, github, etc.

**Comment by vakaobr** [comment]
This python code sample have the url to fetch the commit count using the api https://github.com/sarfraz19/Gitlab/blob/master/%23gitlab_api-data-collector(branches%2Ccommits%2Cetc).py

**Comment by mvisonneau** [comment]
I have implemented this feature as part of the [v0.4.3](https://github.com/mvisonneau/gitlab-ci-pipelines-exporter/releases/tag/v0.4.3) release. It is not enabled by default though, you need to enable `environments` monitoring on your project:

```yaml
project_defaults:
  pull:
    environments:
      enabled: true
```

the metrics which will then be worth of interest are:

`gitlab_ci_environment_behind_commits_count` : _Number of commits the environment is behind given its last deployment_
`gitlab_ci_environment_behind_duration_seconds`: _Duration in seconds the environment is behind the most recent commit given its last deployment_

You can also get the `current_commit_id` and the `latest_commit_id` as a parameter of this metric : `gitlab_ci_environment_information`

For a more concrete example, you can have a look at the [Grafana dashboard regarding environments and deployments](https://grafana.com/grafana/dashboards/13329) which leverages them.

If you have further questions, happy to discuss them in a new issue!
